src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -245,8 +247,10 @@ public function update_item( $request ) {
 		$changes = $this->prepare_item_for_database( $request );
 
 		if ( 'custom' === $template->source ) {
+			$update = true;
 			$result = wp_update_post( wp_slash( (array) $changes ), true );
 		} else {
+			$update = false;
 			$result = wp_insert_post( wp_slash( (array) $changes ), true );
```

Comment:
```suggestion
			$post_before = null;
			$result = wp_insert_post( wp_slash( (array) $changes ), false );
```

Just in case the return of `get_post()` changes for invalid post IDs, let's be explicit to make sure the value is in fact `null`.

:1234: Same change will be needed in `create_item()` but it's outside the diff so I can't add a line comment.